From 21639b51290ff91ed961efa331b826bed434d073 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@lucaweiss.eu>
Date: Tue, 3 Feb 2026 22:20:21 +0100
Subject: [PATCH] init: Add option to install systemd services without systemd
 installed

Allows the sub-packaging of the service in Alpine, where systemd is not
available but packaging of systemd services is allowed (e.g. for
Alpine Linux)
---
 CMakeLists.txt      | 5 +++++
 data/CMakeLists.txt | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 04194ed..7ec12e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,10 +53,15 @@ pkg_check_modules(PROPERTIES_CPP properties-cpp REQUIRED)
 if (ENABLE_TRUST_STORE)
     pkg_check_modules(TRUST_STORE trust-store REQUIRED)
 endif (ENABLE_TRUST_STORE)
+
+option(ENFORCE_SYSTEMD_SERVICE "Enable installation of systemd services" OFF)
 pkg_check_modules(SYSTEMD systemd)
 if(SYSTEMD_FOUND)
     pkg_get_variable(SYSTEMD_SYSTEM_UNITDIR systemd systemd_system_unit_dir)
     pkg_get_variable(SYSTEMD_USER_UNITDIR systemd systemd_user_unit_dir)
+elseif(ENFORCE_SYSTEMD_SERVICE)
+    set (SYSTEMD_SYSTEM_UNITDIR "/usr/lib/systemd/system")
+    set (SYSTEMD_USER_UNITDIR "/usr/lib/systemd/user")
 endif()
 pkg_check_modules(UBUNTU_PLATFORM_HARDWARE_API ubuntu-platform-hardware-api)
 pkg_check_modules(GPSD libgps)
diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index 5973c56..bfb5477 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -45,7 +45,7 @@ install(
   DESTINATION ${CMAKE_INSTALL_DATADIR}/dbus-1/system.d/
 )
 
-if(SYSTEMD_FOUND)
+if(SYSTEMD_FOUND OR ENFORCE_SYSTEMD_SERVICE)
     install(
       FILES ${CMAKE_CURRENT_BINARY_DIR}/lomiri-location-service.service
       DESTINATION ${SYSTEMD_SYSTEM_UNITDIR}
-- 
2.52.0

